client/src/translations/index.tsx

```diff
@@ -0,0 +1,790 @@
+      return false;
+    }
+  } else if (op.operation === "gte") {
+    if (!(value >= op.value)) {
+      return false;
+    }
+  } else if (op.operation === "lt") {
+    if (!(value < op.value)) {
+      return false;
+    }
+  } else if (op.operation === "lte") {
+    if (!(value <= op.value)) {
+      return false;
+    }
+  } else {
+    throw new Error(`Not implemented operation '${op.operation}'`);
+  }
+  return true;
+}
+
+function DocumentsTable({
+  counts,
+  documents,
+}: {
+  counts: Counts;
+  documents: Document[];
+}) {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sort = searchParams.get("sort") || "modified";
+  const sortReverse = JSON.parse(searchParams.get("sortReverse") || "false");
+
+  const page = parseInt(searchParams.get("page") || "1", 10);
+  const pageSize = parseInt(searchParams.get("pageSize") || "20", 10);
+  const filterTitle = searchParams.get("title") || "";
+  const filterURL = searchParams.get("url") || "";
+  const filterDifferences = searchParams.get("differences") || "";
+  const filterDifferencesOperation = getNumericOperation(filterDifferences);
+
+  // https://gist.github.com/jlbruno/1535691/db35b4f3af3dcbb42babc01541410f291a8e8fac
+  function getGetOrdinal(n: number) {
+    const s = ["th", "st", "nd", "rd"];
+    const v = n % 100;
+    return n.toLocaleString() + (s[(v - 20) % 10] || s[v] || s[0]);
+  }
+
+  function TH({ id, title }: { id: string; title: string }) {
+    return (
+      <th
+        onClick={() => {
+          if (sort === id) {
+            setSearchParams(
+              createSearchParams({
+                sort: id,
+                sortReverse: JSON.stringify(!sortReverse),
+              })
+            );
+          } else {
+            setSearchParams(createSearchParams({ sort: id }));
+          }
+        }}
+        className={`sortable ${sort === id ? "active" : ""} ${
+          sort === id && sortReverse ? "reverse" : ""
+        }`}
+      >
+        {title}
+      </th>
+    );
+  }
+
+  function getHighlightedText(text: string, highlight: string) {
+    // Split on highlight term and include term into parts, ignore case
+    const parts = text.split(new RegExp(`(${highlight})`, "gi"));
+    return (
+      <span>
+        {" "}
+        {parts.map((part, i) => (
+          <span
+            key={i}
+            style={
+              part.toLowerCase() === highlight.toLowerCase()
+                ? { fontWeight: "bold" }
+                : {}
+            }
+          >
+            {part}
+          </span>
+        ))}{" "}
+      </span>
+    );
+  }
+
+  function showBriefURL(uri: string) {
+    const [left, right] = uri.split(/\/docs\//, 2);
+    return (
+      <>
+        <span className="url-prefix">{left}/docs/</span>
+        <span className="url-slug">
+          {filterURL ? getHighlightedText(right, filterURL) : right}
+        </span>
+      </>
+    );
+  }
+
+  function showLastModified(doc: Document) {
```

Comment:
Going to sound like a broken record, but it's more true here: I'm not a fan of these nested functions, I think it makes it hard to track what the outer function is doing and which of these is sharing state with it. Imo the default should be not to use them